assembl/graphql/schema.py

```diff
@@ -1324,6 +1334,71 @@ def mutate(root, args, context, info):
         return UpdatePost(post=post)
 
 
+class DeletePost(graphene.Mutation):
+    class Input:
+        post_id = graphene.ID(required=True)
+
+    post = graphene.Field(lambda: Post)
+
+    @staticmethod
+    def mutate(root, args, context, info):
+        discussion_id = context.matchdict['discussion_id']
+
+        user_id = context.authenticated_userid or Everyone
+        discussion = models.Discussion.get(discussion_id)
+
+        post_id = args.get('post_id')
+        post_id = int(Node.from_global_id(post_id)[1])
+        post = models.Post.get(post_id)
+
+        permissions = get_permissions(user_id, discussion_id)
+        allowed = post.user_can(user_id, CrudPermissions.DELETE, permissions)
+        if not allowed:
+            raise HTTPUnauthorized()
+
+        # Same logic as in assembl/views/api2/post.py:delete_post_instance
```

Comment:
I would need to give user_id parameter as well. and return the number_of_extracts removed. I don't like this api. I prefer not to refactor here.